src/Raven.Server/Documents/Handlers/IndexHandler.cs

```diff
@@ -1002,6 +1005,54 @@ await using (var ms = new MemoryStream())
             }
         }
 
+        [RavenAction("/databases/*/indexes/optimize", "POST", AuthorizationStatus.ValidUser, EndpointType.Write, DisableOnCpuCreditsExhaustion = true)]
+        public async Task OptimizeIndex()
+        {
+            using (ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+            {
+                var name = GetQueryStringValueAndAssertIfSingleAndNotEmpty("name");
+                var index = Database.IndexStore.GetIndex(name);
+                if (index == null)
+                    throw new InvalidDataException($"Index '{name}' not found. Cannot perform optimize.");
+                var token = CreateOperationToken();
+                using (index.DrainRunningQueries())
+                {
+                    var result = new CompactionResult(index.Name);
```

Comment:
`CompactionResult`?

Reply:
Left as mock for Operation task

Reply:
Check my below comment:

https://github.com/ravendb/ravendb/pull/15136/files#r1001419051

I think we should have some IndexOptimizeResult here

Reply:
Done